### PR TITLE
Disembiguate generic "Not Found" error reported in JENKINS-45253

### PR DESCRIPTION
**Pull request by i386**
# Description

Improving error messages to help debug [JENKINS-45253](https://issues.jenkins-ci.org/browse/JENKINS-45253).

# Submitter checklist
- [ ] Link to JIRA ticket in description, if appropriate.
- [ ] Change is code complete and matches issue description
- [ ] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [ ] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

### CODE REVIEW

File: blueocean-github-pipeline/src/main/java/io/jenkins/blueocean/blueocean_github_pipeline/GithubScm.java [hunk 1]
```diff
@@ -304,7 +304,7 @@ protected static HttpURLConnection connect(String apiUrl, String accessToken) th
             throw new ServiceException.PreconditionRequired("Github accessToken does not have required scopes. Expected scopes 'user:email, repo'");
         }
         if(status == 404){
-            throw new ServiceException.NotFoundException("Not Found");
+            throw new ServiceException.NotFoundException("Github could not be found. Remote server sent 404.");
```
Comment (michaelneale) [inline blueocean-github-pipeline/src/main/java/io/jenkins/blueocean/blueocean_github_pipeline/GithubScm.java]: phrasing doesn't make sense - "github" can't be sound? is it meant to be github server?

**Comment by michaelneale** [comment]
unfortunately github will just send generic 404s for security reasons, so I don't think this provides any more information for the linked ticket.

**Comment by i386** [comment]
@michaelneale I've updated the message there - should be a bit clearer.

**Comment by michaelneale** [comment]
:bee:

**Pull request merged**